tests/unit_tests/lcls_tools/common/image/test_image_fit.py

```diff
@@ -0,0 +1,18 @@
+import numpy as np
+from lcls_tools.common.data.fit.methods import GaussianModel
+from lcls_tools.common.image.fit import ImageProjectionFit
+
+
+class TestImageProjectionFit:
+    def test_fit(self):
+        test_image = np.zeros((100, 100))
+        test_image[40:60, 40:60] = 255
+
+        image_fit = ImageProjectionFit()
+        result = image_fit.fit_image(test_image)
+
+        assert np.allclose(result.centroid, [50, 50])
+        assert np.allclose(result.rms_size, [8.0347, 8.0347])
+        assert np.allclose(result.total_intensity, 102000.0)
+        assert np.allclose(result.image, test_image)
```

Comment:
Dropping the image test will speed the test up. test_image is 10000 floats, which is a non-trivial comparison.

It's unclear to me that result.image even needs to exist. It seems to be a pass through value. What is the use case of returning it? The problem I have with this is that if image is large, then it will be held in a result, and the caller will have to clear the value to cause the garbage collector to run. A caller that needs to retain the image will likely do that some other way instead of relying on the fitting method to return it.

Reply:
the image needs to be returned for visualizing the results of the fit and dumping the image with the fit results to a single file - IMO fit results and the image should be close to one another when passed around

Reply:
Thanks for the explanation.